nixos/modules/system/boot/loader/grub/install-grub.pl

```diff
@@ -450,7 +450,8 @@ sub addEntry {
 
     # Include second initrd with secrets
     if (-e -x "$path/append-initrd-secrets") {
-        my $initrdName = basename($initrd);
+        # Name the initrd secrets after the system from which they're derived.
+        my $initrdName = basename(Cwd::abs_path("$path"));
```

Comment:
Maybe this variable should now get its name changed, as it is no longer the initrd name but the system name?

Reply:
Good point, I'm updating this.